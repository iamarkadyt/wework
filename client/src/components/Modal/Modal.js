import React from 'react'
import Field from '../Field/Field'
import Overlay from '../Overlay/Overlay'
import './Modal.css'

const Modal = ({
    question,
    onConfirm,
    onDismiss,
    actionColor
}) => {
    const handleDismiss = () => {
        onDismiss()
    }

    const handleConfirm = () => {
        onConfirm()
        onDismiss()
    }

    return (
        <Overlay onBackdropClick={() => handleDismiss()}>
            <div className="Modal-container">
                <p className="question">{question}</p>
                <div className="buttons">
                    <Field
                        type="button"
                        label="No"
                        style={{ width: '100%' }}
                        onClick={() => handleDismiss()} />
                    <Field
                        type="button"
                        label="Yes"
                        style={actionColor
                            ? {
                                backgroundColor: actionColor,
                                borderColor: actionColor,
                                color: 'white',
                                width: '100%'
                            }
                            : null}
                        onClick={() => handleConfirm()} />
                </div>
            </div>
        </Overlay>
    )
}

export default Modal