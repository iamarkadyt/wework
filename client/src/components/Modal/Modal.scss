.Modal-container {
    width: 15rem;
    .question {
        font-size: 1.5rem;
        padding: .5rem;
        padding-bottom: unset;
    }
    .buttons {
        display: grid;
        grid-template-columns: 1fr 1fr;
        grid-column-gap: .3rem;
    }
}