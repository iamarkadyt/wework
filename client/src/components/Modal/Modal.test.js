import React from 'react'
import { shallow } from 'enzyme'
import Modal from './Modal'
import cloneDeep from 'lodash.clonedeep'

let mountedComponent, props

const getMockProps = () => {
  return cloneDeep({
    question: 'Whatsup?',
    onConfirm: jest.fn(),
    onDismiss: jest.fn(),
    actionColor: 'green'
  })
}

const comp = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<Modal {...props} />)
  }
  return mountedComponent
}

describe('Modal', () => {
  beforeEach(() => {
    mountedComponent = undefined
    props = getMockProps()
  })

  describe('props', () => {
    describe('effects', () => {
      describe('actionColor', () => {
        it('if specified, action button takes actionColor for background', () => {
          props.actionColor = 'red'
          const expectedProps = {
            backgroundColor: 'red',
            borderColor: 'red',
            color: 'white',
            width: '100%'
          }
          expect(comp().find('field[label="Yes"]').prop('style')).toEqual(expectedProps)
        })

        it('if not specified, action button does not receive any additional styles', () => {
          delete props.actionColor
          expect(comp().find('field[label="Yes"]').prop('style')).toEqual(null)
        })
      })
    })

    describe('passing', () => {
      describe('Overlay', () => {
        it('receives correct function into onBackdropClick', () => {
          expect(comp().find('Overlay').prop('onBackdropClick')).toBe(props.onDismiss)
        })
      })

      describe('No button', () => {
        it('receives correct function into onClick', () => {
          expect(comp().find('field[label="No"]').prop('onClick')).toBe(props.onDismiss)
        })
      })

      describe('Yes button', () => {
        it('receives correct function into onClick', () => {
          expect(comp().find('field[label="Yes"]').prop('onClick')).toBe(props.onConfirm)
        })
      })
    })
  })

  it('matches snapshot', () => {
    expect(comp()).toMatchSnapshot()
  })
})


