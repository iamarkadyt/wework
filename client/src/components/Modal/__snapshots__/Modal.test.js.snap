// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal matches snapshot 1`] = `
<Overlay
  centered={true}
  onBackdropClick={[MockFunction]}
>
  <div
    className="Modal-container"
  >
    <p
      className="question"
    >
      Whatsup?
    </p>
    <div
      className="buttons"
    >
      <field
        label="No"
        onClick={[MockFunction]}
        style={
          Object {
            "width": "100%",
          }
        }
        type="button"
      />
      <field
        label="Yes"
        onClick={[MockFunction]}
        style={
          Object {
            "backgroundColor": "green",
            "borderColor": "green",
            "color": "white",
            "width": "100%",
          }
        }
        type="button"
      />
    </div>
  </div>
</Overlay>
`;
